functions.php

```diff
@@ -449,20 +449,30 @@ function interconnection_change_publish_button_js() {
  * so we can replace translated content in the loop. This allows
  * us to display English posts if the posts are not translated.
  *
- * This function also removes the sticky post from the homepage loop.
- *
  * @param WP_Query $query The WP_Query instance (passed by reference).
  */
 function interconnection_modify_polylang_query( $query ) {
 	// Query English posts only.
 	if ( function_exists( 'pll__' ) && ! is_admin() && ! is_singular() && $query->is_main_query() ) {
 		$languages = pll_default_language() . ',' . pll_current_language();
 		$query->set( 'lang', $languages );
-
-		// Remove sticky posts from homepage loop.
-		if ( is_home() ) {
-			$query->set( 'post__not_in', get_option( 'sticky_posts' ) ); // phpcs:ignore
-		}
 	}
 }
 //add_action( 'pre_get_posts', 'interconnection_modify_polylang_query' );
+
+/**
+ * Remove the sticky post from the homepage loop.
+ *
+ * @param WP_Query $query The WP_Query instance (passed by reference).
+ */
+function interconnection_remove_homepage_sticky_posts( $query ) {
+	// Get the last added sticky post - last in array.
+	$sticky            = get_option( 'sticky_posts' );
+	$exclude_from_grid = $sticky ? array( $sticky[ count( $sticky ) - 1 ] ) : '';
+
+	if ( ! is_admin() && is_home() && $query->is_main_query() ) {
+		$query->set( 'ignore_sticky_posts', true );
+		$query->set( 'post__not_in', $exclude_from_grid );
```

Comment:
Two questions.
1. should this be set only if `! empty( $exclude_from_grid )`? I understand it won't do anything if the value is empty, but it feels potentially unnecessary
2. Do we actually need the `post__not_in` if we have ignore_sticky_posts? I'd expect the ignore_sticky_posts line to do the trick on it own, if the `post__not_in` is already taken from `sticky_posts`

Reply:
I agree with no. 1. I added the following conditional:

```
if ( ! empty( $exclude_from_grid ) ) {
    $query->set( 'post__not_in', $exclude_from_grid );
}
```

For no. 2, we do need `post__not_in` from what I understand. The `ignore_sticky_posts` parameter only removes the sticky posts functionality, meaning the post won't be moved to the top, but it doesn't remove it from the loop. For that we need `post__not_in`.

Reply:
That makes sense!